crates/tx/src/section.rs

```diff
@@ -442,6 +441,39 @@ pub struct Authorization {
     pub signatures: BTreeMap<u8, common::Signature>,
 }
 
+impl PartialEq for Authorization {
+    fn eq(&self, other: &Self) -> bool {
+        // Deconstruct the two instances to ensure we don't forget any new
+        // field
+        let Authorization {
+            targets,
+            signer: _,
+            signatures,
+        } = self;
+        let Authorization {
+            targets: other_targets,
+            signer: _,
+            signatures: other_signatures,
+        } = other;
+
+        // Two authorizations are equal when they are computed over the same
```

Comment:
This definition could work. But it has the implication that if `Authorization::signer` contains the variant `Signer::PubKeys` and someone tampers with one of the public keys in that list so that `Authorization::verify_signature` can no longer pass, this `PartialEq` implementation would still say that the tampered instance is still equal to the original. No? If so, is this desirable? Given that this function defines a partial equivalence relation, I guess `false` should be returned in those cases where there is insufficient information to check/conclude equality of the signers? No?

Reply:
That's correct. My reasoning was that once we establish the equality of the signature's targets and of the signatures themselves we can safely consider the two `Authorization`s to be the same even if the signers are expressed differently. In case of the signers being straight different (which we cause one of the two instances to fail the signature verification check) I believe we are looking at something else than just equality between the two instances: I guess one should check the integrity of the sections at the byte level using a digest or similar techniques but I don't have a super strong opinion on this (see also my answer to your next question since it's related to this one)